corehq/doctypemigrations/changes.py

```diff
@@ -0,0 +1,29 @@
+from collections import namedtuple
+from couchdbkit import ChangesStream
+import time
+
+CouchChange = namedtuple('CouchChange', ['id', 'rev', 'deleted', 'seq'])
+
+
+def stream_changes(db, since, limit):
+    for change in ChangesStream(db=db, since=since, limit=limit):
+        print change
+        yield CouchChange(
+            id=change['id'], rev=change['changes'][0]['rev'], deleted=change.get('deleted', False),
+            seq=change.get('seq'))
+
+
+def stream_changes_forever(db, since, chunk_size=10000, notify_caught_up=True):
+    last_seq = since
+    while True:
+        changes = stream_changes(db=db, since=last_seq, limit=chunk_size)
+        i = -1
+        change = None
+        for i, change in enumerate(changes):
+            yield change
+        if i + 1 < chunk_size:
+            if notify_caught_up:
+                yield Ellipsis
+            time.sleep(10)
+        if change:
+            last_seq = change.seq
```

Comment:
Just talked this through with @snopoke and I think the conclusion is that to not couple the timing of these two PRs, I'll leave it as is for now and then possibly refactor it after your changes have been merged in.

Also wanted to note that this does one extra thing that's a little different which is to notify the caller whenever it's "caught up" (i.e. less than the full `chunk_size` changes came back) if `notify_caught_up` is set. This bubbles down to the management command so that it prints "All caught up!" periodically, which is a sanity check I wanted for coordinating with deploying the switch-over. The function above this (`stream_changes`), though, could be replaced with a call to `CouchChangeFeed`.